Userland/Libraries/LibGfx/PNGWriter.cpp

```diff
@@ -155,13 +166,13 @@ union [[gnu::packed]] Pixel {
 };
 static_assert(AssertSize<Pixel, 4>());
 
-void PNGWriter::add_IDAT_chunk(Gfx::Bitmap const& bitmap)
+ErrorOr<void> PNGWriter::add_IDAT_chunk(Gfx::Bitmap const& bitmap)
 {
     PNGChunk png_chunk { "IDAT" };
-    png_chunk.reserve(bitmap.size_in_bytes());
+    TRY(png_chunk.reserve(bitmap.size_in_bytes()));
 
     ByteBuffer uncompressed_block_data;
-    uncompressed_block_data.ensure_capacity(bitmap.size_in_bytes() + bitmap.height());
+    TRY(uncompressed_block_data.try_ensure_capacity(bitmap.size_in_bytes() + bitmap.height()));
```

Comment:
Is there a lower bound on the size of uncompressed block data? Can we ensure capacity once and then append unchecked?